http-client/src/main/java/com/facebook/airlift/http/client/jetty/ChunkedBytesContentProvider.java

```diff
@@ -0,0 +1,95 @@
+package com.facebook.airlift.http.client.jetty;
+
+import org.eclipse.jetty.client.util.AbstractTypedContentProvider;
+
+import java.nio.ByteBuffer;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.lang.Math.min;
+import static java.util.Objects.requireNonNull;
+
+public class ChunkedBytesContentProvider
+        extends AbstractTypedContentProvider
+{
+    private static final int DEFAULT_BUFFER_SIZE = 4096;
+
+    private final byte[] bytes;
+    private final int bufferSize;
+
+    public ChunkedBytesContentProvider(byte[] bytes)
+    {
+        this("application/octet-stream", bytes, DEFAULT_BUFFER_SIZE);
+    }
+
+    public ChunkedBytesContentProvider(byte[] bytes, int bufferSize)
+    {
+        this("application/octet-stream", bytes, bufferSize);
+    }
+
+    public ChunkedBytesContentProvider(String contentType, byte[] bytes)
+    {
+        this(contentType, bytes, DEFAULT_BUFFER_SIZE);
+    }
+
+    public ChunkedBytesContentProvider(String contentType, byte[] bytes, int bufferSize)
+    {
+        super(contentType);
+        this.bytes = requireNonNull(bytes, "bytes is null");
+        checkArgument(bufferSize > 0, "bufferSize must be greater then zero: %s", bufferSize);
```

Comment:
```suggestion
        checkArgument(bufferSize > 0, "bufferSize must be greater than zero: %s", bufferSize);
```

Sorry last minute nit